src/main/java/de/rondiplomatico/nds/floodFill.java

```diff
@@ -0,0 +1,271 @@
+package de.rondiplomatico.nds;
+
+import java.awt.Point;
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * Utility class for flood fill algorithm applied to marked tiles deduced from polygon points.
+ * 
+ * @author Andreas Hessenthaler
+ * @since 13.02.2020
+ */
+public class floodFill {
+	int[][] ff;
+	int minX 					= Integer.MAX_VALUE;
+	int maxX 					= Integer.MIN_VALUE;
+	int minY 					= Integer.MAX_VALUE;
+	int maxY 					= Integer.MIN_VALUE;
+	int dimX 					= -1;
+	int dimY 					= -1;
+	int bg 						= 0;
+	int bound 					= 1;
+	int fill 					= bound;
+	int startAtX 				= 0;
+	int startAtY 				= 0;
+	
+	public floodFill(int[][] f, int a, int b, int c, int d) {
+		ff 						= f;
+		minX 					= a;
+		maxX 					= b;
+		minY 					= c;
+		maxY 					= d;
+		dimX 					= maxX - minX + 1;
+		dimY 					= maxY - minY + 1;
+		setStartingPoint();
+	}
+	
+	/*
+	 * Four-way recursive flood fill algorithm
+	 * 
+	 * @param x
+	 * 			the tile x coordinate
+	 * @param y
+	 * 			the tile y coordinate
+	 */
+	public void recursiveFloodFill(int x, int y) {
+		if ((x < 0) || (y < 0) || (x > dimX-1 || (y > dimY-1))) {
+			return;
+		}
+		if (ff[x][y] == bg) {
+			ff[x][y] 			= fill;
+			recursiveFloodFill(x+1, y  ); // right
+			recursiveFloodFill(x,   y+1); // top
+			recursiveFloodFill(x-1, y  ); // left
+			recursiveFloodFill(x,   y-1); // bottom
+		}
+		return;
+	}
+
+	/*
+	 * Eight-way recursive flood fill algorithm
+	 * 
+	 * @param x
+	 * 			the tile x coordinate
+	 * @param y
+	 * 			the tile y coordinate
+	 */
+	public void recursiveFloodFill8(int x, int y) {
+		if ((x < 0) || (y < 0) || (x > dimX-1 || (y > dimY-1))) {
+			return;
+		}
+		if (ff[x][y] == bg) {
+			ff[x][y] 			= fill;
+			recursiveFloodFill(x+1, y  ); // right
+			recursiveFloodFill(x,   y+1); // top
+			recursiveFloodFill(x-1, y  ); // left
+			recursiveFloodFill(x,   y-1); // bottom
+			recursiveFloodFill(x+1, y+1); // top right
+			recursiveFloodFill(x-1, y+1); // top left
+			recursiveFloodFill(x+1, y-1); // bottom right
+			recursiveFloodFill(x-1, y-1); // bottom left
+		}
+		return;
+	}
+
+	/*
+	 * Iterative four-way flood fill algorithm - avoids stack overflow issues
+	 * 
+	 */
+	public void iterativeFloodFill() {
+		Queue<Point> q 			= new LinkedList<Point>();
+		q.add(new Point(startAtX, startAtY));
+		while (!q.isEmpty()) {
+			Point p 			= q.remove();
+			if ((p.x < 0) || (p.y < 0) || (p.x > dimX-1 || (p.y > dimY-1))) {
+				continue;
+			}
+			if (ff[p.x][p.y] == bg) {
+				ff[p.x][p.y] 	= fill;
+				q.add(new Point(p.x+1, p.y  )); // right
+				q.add(new Point(p.x,   p.y+1)); // top
+				q.add(new Point(p.x-1, p.y  )); // left
+				q.add(new Point(p.x,   p.y-1)); // bottom
+			}
+		}
+	}
+
+	/*
+	 * Iterative eight-way flood fill algorithm - avoids stack overflow issues
+	 * 
+	 */
+	public void iterativeFloodFill8() {

```

Comment:
what's the benefit of 4 vs 8 way flood fill?

Reply:
FloodFill8 is potentially faster but can hop across diagonal edges and fill the entire image, .e.g, if you have something like this:

OOOOOO
OOXXOO
OXOOXO
OXXXXO
OOOOOO

Remove flood fill 8 altogether?